### PR TITLE
update eslint for lint typescript code

### LINKED ISSUE #19: Update eslint for lint Typescript code
Actually `.eslintrc` is write for linting ES6 code, rules and recommendations have to be updated for typescript language.

### LINKED ISSUE #20: Add Typescript typecheck executon to CI
The advice you gave me on compatible versions of the types I think is a good idea, I don't have much experience of development library in Typescript, but I understood the meaning of your advice. However, I think that it's good to maintain consistency is the use of Typecheck in the CI. What do you think about?

### LINKED ISSUE #21: Run prettier on whole codebase
I suspect that my Typescript adaptation left a lot of stuff needing prettifying. That's going to produce a bunch of non-substantial but incompatible code changes, so I'd like to do that after merging the TS stuff and before doing any additional development. From there, we can depend on the prettier hook to prettify all committed work before commit.

### PR DESCRIPTION
**Pull request by cimasim89**
Close #19 
Close #20
Close #21

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,8 +7,9 @@
   "scripts": {
     "test": "jest",
     "watch": "jest --watch *.js",
```
Comment (kael-shipman) [inline package.json:9]: I didn't catch this earlier, but we'll probably wanna change this too.

File: src/Types.ts [hunk 2]
```diff
@@ -1,63 +1,101 @@
...
+import * as _ from 'lodash'
+import { isNumeric } from './utils'
 
 // Options to be used to configure the file creator instance as a whole
 export interface GlobalOptions {
-  rowEnd: string;
-  encoding?: BufferEncoding;
-  mode?: number;
-  flag?: string;
+  rowEnd: string
```
Comment (kael-shipman) [inline src/Types.ts:6]: Not sure if this is a setting we can adjust, but preferred typescript syntax has semi-colons at the end of interface lines. See the [intro tutorial](https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes.html) for syntax examples.

While it does appear to compile like this, I would strongly prefer to stick with Typescript standard syntax if possible.

**Comment by kael-shipman** [comment]
Also closes #21

File: src/Types.ts (hunk 2 above)
Comment (cimasim89) [reply src/Types.ts:6]: there are many schools of thought with respect to this point, those who find it more correct with semicolons, those who say they are more productive without... like [here](https://medium.com/@eugenkiss/dont-use-semicolons-in-typescript-474ccfe4bdb3)

**Pull request merged**